fedjax/algorithms/apfl.py

```diff
@@ -239,83 +237,69 @@ def create_eval_for_each_client(model: models.Model):
   """Builds client_init, client_step, client_final for for_each_client."""
 
   def client_init(server_params, client_input):
+    params = interpolate_params(
+      client_input.interpolation_coefficients,
+      client_input.params,
+      server_params)
+
     return {
-      'server_params': server_params,
-      'state': client_input,
+      'params': params,
       'stat': {k: metric.zero() for k, metric in model.eval_metrics.items()}
     }
 
   def client_step(client_step_state, batch):
-    params = interpolate_params(
-      client_step_state['state'].interpolation_coefficients,
-      client_step_state['state'].params,
-      client_step_state['server_params'])
-    
-    stat = models._evaluate_model_step(model, params, batch, client_step_state['stat'])
+    stat = models._evaluate_model_step(
+      model,
+      client_step_state['params'],
+      batch,
+      client_step_state['stat'])
 
     return {
-      'server_params': client_step_state['server_params'],
-      'state': client_step_state['state'],
+      'params': client_step_state['params'],
       'stat': stat
     }
 
   def client_final(server_params, client_step_state):
     del server_params
-
-    return jax.tree_util.tree_map(
-      lambda x: x.result(),
-      client_step_state['stat'],
-      is_leaf=lambda v: isinstance(v, metrics.Stat))
+    return client_step_state['stat']
 
   return for_each_client.for_each_client(client_init, client_step, client_final)
 
 
 def eval_adaptive_personalized_federated_learning(
     model: models.Model,
-    client_batch_hparams: client_datasets.ShuffleRepeatBatchHParams
-) -> federated_algorithm.FederatedAlgorithm:
+    client_batch_hparams: client_datasets.PaddedBatchHParams,
+) -> Callable[
+  [ServerState, Sequence[Tuple[federated_data.ClientId, client_datasets.ClientDataset]]],
+  Sequence[Tuple[federated_data.ClientId, Mapping[str, jnp.array]]]
+]:
   """Evaluates adaptive personalized federated learning.
 
   Args:
     model: Model
-    client_batch_hparams: Hyperparameters for batching client dataset for train.
+    client_batch_hparams: Hyperparameters for batching client dataset for eval.
+    aggregate: Flag whether eval metrics should be aggregated over clients.
```

Comment:
This arg description for `aggregate` looks like it was left in by accident since I don't see it in the args list.